ads/google/a4a/utils.js

```diff
@@ -88,13 +88,10 @@ export const TRUNCATION_PARAM = {name: 'trunc', value: '1'};
 export function isGoogleAdsA4AValidEnvironment(win) {
   const supportsNativeCrypto = win.crypto &&
       (win.crypto.subtle || win.crypto.webkitSubtle);
-  // Note: Theoretically, isProxyOrigin is the right way to do this, b/c it
-  // will be kept up to date with known proxies.  However, it doesn't seem to
-  // be compatible with loading the example files from localhost.  To hack
-  // around that, just say that we're A4A eligible if we're in local dev
-  // mode, regardless of origin path.
+  const googleCdnProxyRegex =
+      /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org/;
   return supportsNativeCrypto &&
-      (isProxyOrigin(win.location) || getMode(win).localDev ||
+      (googleCdnProxyRegex.test(win.location.origin) || getMode(win).localDev ||
```

Comment:
Good catch. We can't add a trailing slash though, that won't match an origin; we need a trailing $. Also, that regex was copy-pasted from `isProxyOrigin` so other code has the same bug.